### ISSUE TITLE
Server Malfunction on Incorrect Bytecode Submission to `hardhat_setCode` RPC

### ISSUE THREAD

**Issue by mpopovac-txfusion** [matter-labs/era-test-node#252]
#### 📝 Description

When incorrect bytecode is submitted to the `hardhat_setCode` RPC endpoint, the zkSync Era in-memory node experiences a critical malfunction that affects the entire server. This issue arises after the first attempt with wrong bytecode, leading to a complete server breakdown for all functionalities, not just the `hardhat_setCode` endpoint. Subsequent attempts to use any server functionality result in errors until the server is restarted.

#### 🔄 Reproduction Steps

1. Set up the zkSync Era in-memory node.
2. Invoke the `hardhat_setCode` RPC endpoint with incorrect bytecode.
3. Observe the error and try invoking any server functionality again.
4. Notice the persistent error across all server functionalities, requiring a server restart.

#### 🤔 Expected Behavior

The expectation is that if incorrect bytecode is submitted, the server should return an error message but continue to function normally, allowing for subsequent calls to any server functionality without requiring a server restart.

#### 😯 Current Behavior

After submitting incorrect bytecode once, the entire server malfunctions. This affects not just the `hardhat_setCode` endpoint, but all server functionalities. The error persists until the server is restarted. 

#### 🖥️ Environment

- **Rust version**: rustc 1.70.0
- **Operating System & Version**: macOS, 13.3.1, apple chip M2

#### 📎 Log Output

```
thread 'tokio-runtime-worker' panicked at 'Bytes must be divisible by 32 to split into chunks', /home/runner/.cargo/git/checkouts/zksync-era-657998444dcd9ed9/3c669e7/core/lib/utils/src/convert.rs:67:5
note: run with RUST_BACKTRACE=1 environment variable to display a backtrace
10:38:50 ERROR failed setting code: failed acquiring lock: PoisonError { .. }
10:41:00 ERROR failed setting code: failed acquiring lock: PoisonError { .. }
10:41:04 ERROR failed setting code: failed acquiring lock: PoisonError { .. }
10:50:35 ERROR failed setting code: failed acquiring lock: PoisonError { .. }
```

**Comment by MexicanAce** [comment]
Do you mean `hardhat_setStorageAt`? I don't believe we support `hardhat_setStorageAt`, see [SUPPORTED_APIS](https://github.com/matter-labs/era-test-node/blob/main/SUPPORTED_APIS.md) for our full list of supported endpoints.

Can you double check that you're pointed to the `era_test_node` binary? That endpoint doesn't exist in our namespace/interface, so I would've expected a different error than what you're seeing.

**Comment by mpopovac-txfusion** [comment]
I've updated the issue. It is `hardhat_setCode`.

**Comment by bonedaddy** [comment]
Also running into this issue when using test node version `0.1.0-alpha.19`

**Comment by MexicanAce** [comment]
@bonedaddy @mpopovac-txfusion This issue should now be resolved with yesterday's release of `0.1.0-alpha.24`
